parsers/regex/regex.go

```diff
@@ -0,0 +1,165 @@
+// Package regex consumes logs given user-defined regex format for lines
+
+/* Example usage (from honeytail root)
+  		go install . && \
+ 			honeytail -p regex -k $HONEYTAIL_WRITEKEY \
+			-f /private/var/system.log \
+			--dataset 'MY_TEST_DATASET' \
+			--backfill \
+			--regex.line_regex="(?P<time>\d{2}:\d{2}:\d{2})" \
+			--regex.timefield="time" \
+			--regex.time_format="%H:%M:%S"
+*/
+
+package regex
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+type Options struct {
+	LineRegex       string `long:"line_regex" description:"a regular expression with named capture groups representing the fields you want parsed"`
+	TimeFieldName   string `long:"timefield" description:"Name of the field that contains a timestamp"`
+	TimeFieldFormat string `long:"time_format" description:"Timestamp format to use (strftime and Golang time.Parse supported)"`
+
+	NumParsers int `hidden:"true" description:"number of regex parsers to spin up"`
+}
+
+type Parser struct {
+	conf       Options
+	lineParser parsers.LineParser
+}
+
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+
+	// Regex can't be blank
+	if p.conf.LineRegex == "" {
+		logrus.Debug("LineRegex is blank")
+		return errors.New("LineRegex is blank")
```

Comment:
Small request: this error gets bubbled up to users if they don't specify the `--regex.line_regex` flag. Can we make it a little clearer? Right now, running honeytail without specifying a regex prints:
```
$ honeytail -f - -k $WRITEKEY -d test -p regex
INFO[0000] Starting honeytail                           
FATA[0000] err initializing parser module                err=LineRegex is blank parser=regex
```
which would probably be a bit opaque to the typical user.